### PR TITLE
Add documentation on output of Aryn Partitioning Service

### PR DESCRIPTION
**Pull request by MarkLindblad**

### CODE REVIEW

File: docs/source/aryn_cloud/aryn_partitioning_service.md [hunk 1]
```diff
@@ -118,3 +118,45 @@ You can use multiple options at the same time like in the example below:
 export ARYN_API_KEY="PUT API KEY HERE"
 curl "https://api.aryn.cloud/v1/document/partition" -o "document.json" -H "Authorization: Bearer $ARYN_API_KEY" -F "pdf=@document.pdf" -F "options={\"extract_table_structure\": true, \"threshold\": 0.2}";
 ```
+
+## Output Format
+
+The output of the Aryn Partitioning Service is json.
```
Comment (jonfritz) [inline docs/source/aryn_cloud/aryn_partitioning_service.md:124]: JSON

File: docs/source/aryn_cloud/aryn_partitioning_service.md [hunk 2]
```diff
@@ -118,3 +118,45 @@ You can use multiple options at the same time like in the example below:
...
+
+```text
+{ "status": in-progress updates,
+  "error": any errors encountered,
+  "elements": a list of elements }
+```
```
Comment (jonfritz) [inline docs/source/aryn_cloud/aryn_partitioning_service.md:130]: Do we need to explain what an element is?

File: docs/source/aryn_cloud/aryn_partitioning_service.md [hunk 3]
```diff
@@ -118,3 +118,45 @@ You can use multiple options at the same time like in the example below:
...
+
+### Element Format
+
+{
+```text
+"type": one of the element types below,
+```
+
+Element Types: "N/A","Caption","Footnote","Formula","List-item","Page-footer","Page-header","Picture","Section-header","Table","Text","Title"
```
Comment (jonfritz) [inline docs/source/aryn_cloud/aryn_partitioning_service.md:139]: Should we explain what these are in a table, where the first column is type and second column is description of the classification?

File: docs/source/aryn_cloud/aryn_partitioning_service.md [hunk 4]
```diff
@@ -118,3 +118,45 @@ You can use multiple options at the same time like in the example below:
...
+
+```text
```
Comment (jonfritz) [inline docs/source/aryn_cloud/aryn_partitioning_service.md:141]: Make each of these a subheading, so it's easy to see each part of "element" when reading.

File: docs/source/aryn_cloud/aryn_partitioning_service.md [hunk 5]
```diff
@@ -118,3 +118,45 @@ You can use multiple options at the same time like in the example below:
...
+"bbox": coordinates of the bounding box around the element contents,
+```
+Takes the format `[x1, y1, x2, y2]` where it is a proportion of how far down the screen the element is.
+```text
+"properties":
+    { "score": confidence (between 0 and 1, with 1 being the most confident),
+      "page_number": 1-indexed page number the element occurs on },
+```
```
Comment (jonfritz) [inline docs/source/aryn_cloud/aryn_partitioning_service.md:149]: Explain what the confidence is for

File: docs/source/aryn_cloud/aryn_partitioning_service.md [hunk 6]
```diff
@@ -118,3 +118,45 @@ You can use multiple options at the same time like in the example below:
...
+
+```text
+"text_representation": text associated with this element,
+```
+
+Text elements contain ‘\n’ when the text includes a line return
+
+When `extract_images` is set to True, Pictures include a binary_representation tag which contains a base64 encoded ppm image file of the pdf cropped to the bounds of the detected image. When `extract_images` is false, the bounding box of the Picture is still returned.
```
Comment (jonfritz) [inline docs/source/aryn_cloud/aryn_partitioning_service.md:157]: Should we explain the table output as well, or is that pending on Ben?

**Review by jonfritz — DISMISSED** [review]
Let's add this as a new section under Aryn Partitioning Service, where there is a carrot on the service page to get to this nested page.

File: docs/source/aryn_cloud/aryn_partitioning_service.md (hunk 6 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.md]: This is pending on @bsowell

File: docs/source/aryn_cloud/aryn_partitioning_service.md (hunk 1 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.md]: Done.

**Comment by MarkLindblad** [comment]
> Let's add this as a new section under Aryn Partitioning Service, where there is a carrot on the service page to get to this nested page.

Done.

File: docs/source/aryn_cloud/aryn_partitioning_service.md (hunk 4 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.md]: Done.

File: docs/source/aryn_cloud/aryn_partitioning_service.md (hunk 5 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.md]: Done.

File: docs/source/aryn_cloud/aryn_partitioning_service.md (hunk 3 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.md]: Done.

File: docs/source/aryn_cloud/aryn_partitioning_service.md (hunk 2 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.md]: Done.

File: docs/source/aryn_cloud/aps_output.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+| Text    | Regular Text | 
+| Caption | Description of an image or table |
+| Footnote | Small text found near the bottom of the page |
+| Formula | LaTeX or similar mathematical expression |
+| List-item | Part of a list |
+| Page-footer | Small text at bottom of page |
+| Page-header | Small text at top of page |
+| Image | A Picture or diagram. When `extract_images` is set to `true`, this element includes a `binary_representation` tag which contains a base64 encoded ppm image file. When `extract_images` is false, the bounding box of the Image is still returned. |
+| Section-header | Medium-sized text marking a section. |
+| Table | A grid of text. See the `extract_table_structure` option to extract information from the table rather than just detecting its presence. |
```
Comment (AbhijitP-009) [inline docs/source/aryn_cloud/aps_output.md:63]: isn't this returned as lower case "table" for some reason?

File: docs/source/aryn_cloud/aps_output.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+### Bounding Box
+
+```text
+"bbox": coordinates of the bounding box around the element contents
+```
+Takes the format `[x1, y1, x2, y2]` where each coordinate is a proportion of how far down or across the screen the element is.
```
Comment (AbhijitP-009) [inline docs/source/aryn_cloud/aps_output.md:70]: What does "proportion of how far down or across the screen" mean?

File: docs/source/aryn_cloud/aryn_partitioning_service.rst [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+=========================
+
+.. toctree::
+   :caption: Aryn Cloud
+   :maxdepth: 1
+   :hidden:
+
+   ./aps_output.md
+
+You can use the Aryn Partitioning Service to segment PDF's into labeled bounding boxes identifying titles, tables, table rows and columns, images, and regular text. Bounding boxes are returned as JSON with their associated text for easy use. It leverages `Aryn's purpose-built AI model for document segmentation and labelling <https://huggingface.co/Aryn/deformable-detr>`_ that was trained using DocLayNet – an open source, human-annotated document layout segmentation dataset containing tens of thousands of pages from a broad variety of document sources.
```
Comment (AbhijitP-009) [inline docs/source/aryn_cloud/aryn_partitioning_service.rst:11]: PDF's -> PDFs

File: docs/source/aryn_cloud/aryn_partitioning_service.rst [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+
+If you'd like to experiment with the service, you can use a UI to visualize how your documents are partitioned in the `Aryn Playground <https://www.play.aryn.ai/partitioning>`_. Also, you can view `a notebook with sample code using the service <https://github.com/aryn-ai/sycamore/blob/main/notebooks/ArynPartitionerPython.ipynb>`_ and `a notebook using the service with Langchain <https://github.com/aryn-ai/sycamore/blob/main/notebooks/ArynPartitionerWithLangchain.ipynb>`_.
```
Comment (AbhijitP-009) [inline docs/source/aryn_cloud/aryn_partitioning_service.rst:14]: should it say "you can use the UI in the Playground<...> to visualize how your document will be parititoned"

File: docs/source/aryn_cloud/aryn_partitioning_service.rst [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+
+    # Produce a pandas DataFrame representing one of the extracted tables
+    table_elements = [elt for elt in data['elements'] if elt['type'] == 'table']
+    dataframe = table_elem_to_dataframe(table_elements[0])
+
+    # Draw the detected bounding boxes on the pages. requires poppler
+    images = draw_with_boxes("mydocument.pdf", data)
+
++++++++++++++++++++++++++++++++++++++++++
+Using with Sycamore's Partition transform
```
Comment (AbhijitP-009) [inline docs/source/aryn_cloud/aryn_partitioning_service.rst:70]: "Using with" sounds off here

File: docs/source/aryn_cloud/aryn_partitioning_service.rst [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+
+.. code:: python
+
+    aryn_api_key = "PUT API KEY HERE"
+
+    ctx = sycamore.init()
+    pdf_docset = context.read.binary(work_dir, binary_format="pdf")
+    partitioned_docset = pdf_docset.partition(ArynPartitioner(aryn_api_key=aryn_api_key))
+
+Alternatively, we can store our Aryn API key at ``~/.aryn/config.yaml`` like so:
```
Comment (AbhijitP-009) [inline docs/source/aryn_cloud/aryn_partitioning_service.rst:83]: "we can store our Aryn API key in" ?

File: docs/source/aryn_cloud/aryn_partitioning_service.rst [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+We can extract the table structure from our document with the following command.
+
+.. code:: bash
+
+    export ARYN_API_KEY="PUT API KEY HERE"
+    curl -s -N -D headers "https://api.aryn.cloud/v1/document/partition" -H "Authorization: Bearer $ARYN_API_KEY" -F "pdf=@document.pdf" -F 'options={"extract_table_structure": true}' | tee document.json
+
+The available options are listed below:
+
+* ``threshold``: The threshold to use for accepting the models predicted bounding boxes. A lower value will include more objects, but may have overlaps, a higher value will reduce the number of overlaps, but may miss legitimate objects. It defaults to ``0.4``.
```
Comment (AbhijitP-009) [inline docs/source/aryn_cloud/aryn_partitioning_service.rst:136]: "the model's predicted bounding boxes"

File: docs/source/aryn_cloud/aryn_partitioning_service.rst [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+* ``use_ocr``: If ``true``, the partitioner uses OCR to extract text from the PDF. It defaults to ``false``, where the partitioner attempts to directly extract the text from the underlying PDF in the bounding box. It currently uses Tesseract for extraction.
+* ``extract_table_structure``: If ``true``, the partitioner runs a separate table extraction model to extract cells from regions of the document identified as tables.
```
Comment (AbhijitP-009) [inline docs/source/aryn_cloud/aryn_partitioning_service.rst:138]: "separate table extraction model" is a bit confusing here. Separate from what?

File: docs/source/aryn_cloud/aryn_partitioning_service.rst [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+
+.. code:: bash
+
+    export ARYN_API_KEY="PUT API KEY HERE"
+    curl -s -N -D headers "https://api.aryn.cloud/v1/document/partition" -H "Authorization: Bearer $ARYN_API_KEY" -F "pdf=@document.pdf" -F 'options={"extract_table_structure": true}' | tee document.json
+
+The available options are listed below:
+
+* ``threshold``: The threshold to use for accepting the models predicted bounding boxes. A lower value will include more objects, but may have overlaps, a higher value will reduce the number of overlaps, but may miss legitimate objects. It defaults to ``0.4``.
+* ``use_ocr``: If ``true``, the partitioner uses OCR to extract text from the PDF. It defaults to ``false``, where the partitioner attempts to directly extract the text from the underlying PDF in the bounding box. It currently uses Tesseract for extraction.
```
Comment (AbhijitP-009) [inline docs/source/aryn_cloud/aryn_partitioning_service.rst:137]: Could we rephrase the second sentence? Maybe "It defaults to ``false``, where the partitioner attempts to directly extract the text from the underlying PDF using Tesseract." Also what exactly is Tesseract?

**Review by AbhijitP-009 — COMMENTED** [review]
Left some comments

File: docs/source/aryn_cloud/aps_output.md (hunk 7 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aps_output.md]: Now changed in the docs to lowercase.

File: docs/source/aryn_cloud/aryn_partitioning_service.rst (hunk 9 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.rst]: Done.

File: docs/source/aryn_cloud/aryn_partitioning_service.rst (hunk 10 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.rst]: Done.

File: docs/source/aryn_cloud/aryn_partitioning_service.rst (hunk 11 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.rst]: Fixed.

File: docs/source/aryn_cloud/aryn_partitioning_service.rst (hunk 12 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.rst]: Done.

File: docs/source/aryn_cloud/aryn_partitioning_service.rst (hunk 13 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.rst]: Fixed.

File: docs/source/aryn_cloud/aryn_partitioning_service.rst (hunk 14 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.rst:138]: Separate from the segmentation/partitioning model.

Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.rst]: Added context.

File: docs/source/aryn_cloud/aryn_partitioning_service.rst (hunk 15 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aryn_partitioning_service.rst]: Rephrased.

File: docs/source/aryn_cloud/aps_output.md (hunk 8 above)
Comment (MarkLindblad) [reply docs/source/aryn_cloud/aps_output.md]: Added more explanation.

**Review by AbhijitP-009 — APPROVED** [review]

**Pull request merged**